framework/src/problems/FEProblemBase.C

```diff
@@ -3705,6 +3705,9 @@ FEProblemBase::execute(const ExecFlagType & exec_type)
       _displaced_problem->setCurrentlyComputingJacobian(true);
   }
 
+  if (exec_type != EXEC_INITIAL)
```

Comment:
how's the order during INITIAL?

Reply:
Roughly like so:
```
if (!_app.isRecovering())
{
  ...
  execute(EXEC_INITIAL);
  compute initial material Loop
}
executeControls(EXEC_INITIAL);
```

So, if I understand correctly, `execute` is optional during app recovering. When starting up, there needs to be an extra material loop. So this prevents evaluating the control system twice during `execute_on = initial`